crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -259,6 +263,7 @@ impl CumulocityConverter {
             downloader_sender,
             pending_operations: HashMap::new(),
             command_id,
+            custom_entities: VecDeque::with_capacity(CUSTOM_ENTITY_CACHE_SIZE),
```

Comment:
I see no value added by this cache of unknown entities and I would simply notify an error for each and every message received for an entity that is not already registered and cannot be automatically registered.

The code of the c8y converter is already is already too complex and doesn't need this extra complexity. Furthermore, this may confuse the user: sending a first message, noticing the error, retrying to understand what is going on, noticing no more errors, wrongly assuming everything is fine.

Reply:
I'll let @reubenmiller comment on whether he's okay to remove it. It definitely adds complexity for a minor enhancement, I agree, but it was done on an explicit request to avoid flooding the logs and the `te/error` topics when there are a lot of telemetry messages processed back-to-back with the same unregistered topic id. This will mostly happen during the startup of the mapper, where all persistent telemetry messages queued when the mapper was down, are delivered together.

Reply:
Yes we can remove the "unknown entity" store, and in place just log (somewhere) on the DEBUG level so that we don't cause any undue spamming on production systems, but still provide debugging ability in development environments.

Reply:
I'm still not fully convinced about logging such a critical user error at DEBUG level, as users typically enable DEBUG level to get more insights into what's going on, when they have already spotted something going wrong. Discovering an error itself from a DEBUG log is quite unusual and finding it from a sea of debug statements is going to be a near impossible task.

Even on the log spamming issue, I really don't see how this can happen in production (once #2428 is fixed, of course). Someone using a custom topic scheme without explicitly registering it sounds like a basic user mistake during the development phase, which would/should be caught before that deployment goes into production. During that development phase, logging user mistakes as ERRORs would be far more beneficial to the user than making it DEBUG. And spamming logs for basic mistakes doesn't sound like a big issue in a dev environment as the same spamming would happen with any other trivial mistakes that they make as well.

But, as you pointed out offline, since the error is also reported to `te/errors`, that log statement missing may not be such a big deal for "error discoverability". So, to avoid stretching this PR further on such a trivial matter, I'm okay with the proposed change.